src/main/resources/static/js/gw.workflow.js

```diff
@@ -1055,130 +1059,179 @@ GW.workflow = {
 			"  </thead> "+
 			"  <tbody> ";
 
-			
+
 			for(var i=0;i<msg.length;i++){
-				
+
 				var status_col = GW.history.getWorkflowStatusCol(msg[i].id, msg[i].status);
-				
+
 				content += "    <tr> "+
 					"      <td>"+msg[i].name+"</td> "+
 					"      <td>"+msg[i].begin_time+"</td> "+
 					"      <td>"+msg[i].end_time+"</td> "+
 					status_col+
 					"      <td><a href=\"javascript: GW.workflow.getHistoryDetails('"+msg[i].id+"')\">Check</a></td> "+
 					"    </tr>";
-				
+
 			}
-			
+
 			content += "</tbody></table></div>";
 
 			content += '<div class="modal-footer">' +
 	        				"	<button type=\"button\" class=\"btn btn-outline-secondary general-cancel-btn\">Cancel</button>"+
 	        				'</div>';
 
-			
+
 			var frame = GW.process.createJSFrameDialog(800, 640, content, "History")
 
 			$(".general-cancel-btn").click(function(){
-				
+
 				frame.closeFrame();
-				
+
 			});
-			
+
 		}).fail(function(jxr, status){
-			
+
 			console.error(status);
-			
+
 		});
-		
+
 	},
-	
-	
+
+
 
 	stop: function(workflow_history_id){
 
 		console.log("Send stop request to stop the running workflow");
-			
+
 		$.ajax({
-			
+
 			url: "stop",
-			
+
 			method: "POST",
-			
+
 			data: {type: "workflow", id: workflow_history_id}
-			
+
 		}).done(function(msg){
-			
+
 			msg = $.parseJSON(msg);
-			
+
 			console.log("stopping workflow is called");
 
 			if(msg.ret=="stopped"){
-				
+
 				$("#stopbtn_" + workflow_history_id).html("<span class=\"text-success\">Stopped</span>");
-				
+
 				$("#stopbtn_" + workflow_history_id).prop("onclick", null).off("click");
-				
+
 //					<span id=\"status_"+msg[i].id+"\" class=\"label label-warning\">Pending</span>
-				
+
 				$("#status_" + workflow_history_id).html("<span class=\"label label-default\">Stopped</span>");
 
 				if(workflow_history_id == GW.workflow.history_id){
 
 					GW.monitor.switchPlayButton(true);
 
 				}
-				
+
 			}else{
 
 				alert("Fail to stop.");
 
 			}
-			
+
+		});
+
+
+	},
+
+	checkpoint: function(workflowId) {
```

Comment:
this function name should be `get_checkpoint` or `list_checkpoints_of_workflow`. Also, I am not sure this is necessary. Because we only need one restore button for each workflow history. That is it. No need to query checkpoints because workflow history already has it. 

In other words, we don't want `checkpoints` function. We only save checkpoints automatically when people run the workflow.